### PR TITLE
#211 Year Specific Source Home Care R

### LINKED ISSUE #231: Convert SPSS Home Care (year specific)

### PR DESCRIPTION
**Pull request by catherineholland1**
Closes #231 

Conversion of the Year Specific Home Care Extract

Includes a couple of functions created in the Care Home PR #223 - I have copied and pasted into this branch to run. 

I have also changed the latest update to `Jun_2022` to get the new costs variables read in and updated into the code

Package reference index updated also

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -18,6 +23,12 @@ export(create_hb_test_flags)
 export(create_hscp_test_flags)
 export(create_monthly_beddays)
 export(create_monthly_costs)
+<<<<<<< refs/remotes/origin/main-R
+=======
+export(create_person_id)
+export(date_to_numeric)
+export(dd_period)
+>>>>>>> source home care conversion
```
Comment (Moohan) [inline NAMESPACE:31]:
```suggestion
export(create_person_id)
export(date_to_numeric)
export(dd_period)
```
Conflict markers to be fixed.

File: NAMESPACE [hunk 2]
```diff
@@ -3,13 +3,18 @@
...
 export(convert_date_to_numeric)
 export(convert_eng_gpprac_to_dummy)
 export(convert_fyyear_to_year)
+<<<<<<< refs/remotes/origin/main-R
 export(convert_hb_to_hbnames)
 export(convert_hscp_to_hscpnames)
 export(convert_numeric_to_date)
+=======
+export(convert_sending_location_to_lca)
+>>>>>>> source home care conversion
```
Comment (Moohan) [inline NAMESPACE:17]:
```suggestion
export(convert_hb_to_hbnames)
export(convert_hscp_to_hscpnames)
export(convert_numeric_to_date)
export(convert_sending_location_to_lca)
```

**Review by Moohan — COMMENTED** [review]
Just fixing some conflict markers.

File: Production_scripts/Process_extracts/extract_home_care.R [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+library(dbplyr)
+library(createslf)
+
+
+year <- check_year_format("1920")
+
+
+# Read in data---------------------------------------
+
+source_hc_data <- readr::read_rds(get_sc_hc_episodes_path(update = "Jun_2022")) %>%
```
Comment (Jennit07) [inline Production_scripts/Process_extracts/extract_home_care.R:23]:
```suggestion
source_hc_data <- readr::read_rds(get_sc_hc_episodes_path(update = latest_update())) %>%
```
Should this have the latest_update function here? We have just finished the latest update so i will update these functions to reflect this

File: Production_scripts/Process_extracts/extract_home_care.R [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+      hc_service == 2 ~ "HC-Per",
+      TRUE ~ "HC-Unknown"
+    )
+  ) %>%
+  # age
+  compute_mid_year_age(year, dob) %>%
+  # person_id
+  create_person_id() %>%
+  # compute lca variable from sending_location
+  mutate(lca = convert_sending_location_to_lca(sending_location))
```
Comment (Jennit07) [inline Production_scripts/Process_extracts/extract_home_care.R:70]: Ive pushed some changes on this block of code as it wasn't working quite right. I think that should be it now. I was getting errors with the year and functions converting the year

File: Production_scripts/Process_extracts/extract_home_care.R [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  rename(
+    hc_costs_q1 = paste0("hc_cost_", convert_fyyear_to_year(year), "Q1"),
+    hc_costs_q2 = paste0("hc_cost_", convert_fyyear_to_year(year), "Q2"),
+    hc_costs_q3 = paste0("hc_cost_", convert_fyyear_to_year(year), "Q3"),
+    hc_costs_q4 = paste0("hc_cost_", convert_fyyear_to_year(year), "Q4")
+  ) %>%
+  # remove cost variables not from current year
+  select(-(contains("hc_cost_2"))) %>%
+  # create cost total net
+  mutate(cost_total_net = rowSums(across(contains("hc_cost_q"))))
```
Comment (Jennit07) [inline Production_scripts/Process_extracts/extract_home_care.R:102]: Something is not quite right when calculating the hours and costs. If you run to line 69 the hours and cost values are present. After running this block of code the hours and costs now return 0?

**Review by Jennit07 — CHANGES_REQUESTED** [review]
A few changes suggested and a potential bug with the hours and costs

File: Production_scripts/Process_extracts/extract_home_care.R (hunk 5 above)
Comment (catherineholland1) [reply Production_scripts/Process_extracts/extract_home_care.R:102]: it seems that the costs and hours for 2019/20 seem to be 0? could this be an issue with the data being read in not being populated with values?

File: Production_scripts/Process_extracts/extract_home_care.R (hunk 3 above)
Comment (catherineholland1) [reply Production_scripts/Process_extracts/extract_home_care.R]: latest_update() has not been updated yet to be `Jun_2022` so costs are not read in

File: Production_scripts/Process_extracts/extract_home_care.R (hunk 5 above)
Comment (Moohan) [reply Production_scripts/Process_extracts/extract_home_care.R:100]: I think this was caused by the selection near the top being wrong. i.e. it wasn't selecting the right data for the given year

**Comment by Moohan** [comment]
@Jennit07 can you review please? 

I'm happy with it now but also made a lot of changes myself.

**Review by Jennit07 — APPROVED** [review]
Looks good to me too. I asked one question on teams about how we should be styling the save outfile at the end and generally we will look at this case by case and decide whether to have parameters as a list or one line. 

Can use margin by going to `global options > code > display` as a guide. We can also use the built in code formatter by highlighting the code and using `ctrl + shift + A`

Ready to merge

**Pull request merged**